File: comparison/results/haiku/codecraft/humaneval/HumanEval_33/java/solution.java
public static ArrayList<Long> sortThird(ArrayList<Long> l) {
    List<Long> thirdIndices = new ArrayList<>();
    for (int i = 0; i < l.size(); i++) {
        if (i % 3 == 0) {
            thirdIndices.add(l.get(i));
        }
    }
    
    Collections.sort(thirdIndices);
    
    ArrayList<Long> result = new ArrayList<>();
    int thirdIndex = 0;
    for (int i = 0; i < l.size(); i++) {
        if (i % 3 == 0) {
            result.add(thirdIndices.get(thirdIndex++));
        } else {
            result.add(l.get(i));
        }
    }
    
    return result;
}

File: comparison/results/haiku/codecraft/humaneval/HumanEval_33/typescript/solution.ts
function sort_third(l: number[]): number[] {
    const thirdValues = l.filter((_, i) => i % 3 === 0).sort((a, b) => a - b);
    const result = [...l];
    let sortedIndex = 0;
    for (let i = 0; i < result.length; i++) {
        if (i % 3 === 0) {
            result[i] = thirdValues[sortedIndex++];
        }
    }
    return result;
}
